### PR TITLE
Encourage temporary MFA-authenticated credentials on the command-line

### PR DESCRIPTION
**Pull request by pingles**
Most developers are familiar with using the Access and Secret Keys downloaded from the AWS Console. Security can be improved by:
1. Requiring MFA be used when performing privileged API operations (`ec2:TerminateInstances`, `rds:Delete*` etc.).
2. Using temporary credentials for everything except requesting temporary keys.

You can obviously just use the AWS CLI tools to achieve this (by invoking the different operations) but we wrote a little CLI tool to make it easier to do.

### CODE REVIEW

**Comment by jlevy** [comment]
👍 These seems like good additions.

That looks like a useful tool — though I'm a little hesitant to link to a GitHub tool that's brand new. Would you be up for committing the tips without the tool until we get more feedback on the tool? Or sharing a bit more on its maturity? Glad to chat more in Slack #security too!

@ThanosBaskous @richadams @ehammond thoughts/review?

File: README.md [hunk 1]
```diff
@@ -550,6 +550,7 @@ We cover security basics first, since configuring user accounts is something you
 	-	You should always use MFA, and the sooner the better — enabling it when you already have many users is extra work.
 	-	Unfortunately it can’t be enforced in software, so an administrative policy has to be established.
 	-	Most users can use the Google Authenticator app (on [iOS](https://itunes.apple.com/us/app/google-authenticator/id388497605) or [Android](https://play.google.com/store/apps/details?id=com.google.android.apps.authenticator2)) to support two-factor authentication. For the root account, consider a hardware fob.
+	- Instead of using long-term API keys when using the AWS APIs from the commandline it's better to use temporary keys from [Security Token Service](http://docs.aws.amazon.com/STS/latest/APIReference/Welcome.html). By applying a policy to users its also a good idea to require MFA authentication for any privileged API calls. Tools like [stscreds](https://github.com/uswitch/stscreds) can help this.
```
Comment (jlevy) [inline README.md]: Nits: commandline -> command-line
its also -> it’s also

**Comment by pingles** [comment]
Totally understand wrt it being new. It's possible to achieve the same directly using the AWS command-line tools (albeit a little more fiddly- which is where the tool came in).

I could update with an example of achieving the same through the AWS command-line and example policy if that helps? Also happy to wait for other people to take a look at the tool first.

**Comment by ehammond** [comment]
Since I was @ mentioned... I totally support the recommendation to use temporary credentials and MFA where possible. I also agree that non-Amazon software recommendations should be avoided, unless they are completely accepted by the marketplace (e.g., "serverless"), especially when it relates to security management.

**Comment by jlevy** [comment]
@pingles Given the feedback, could you remove the tool for now and handle the copy editing tweaks? Of you do that glad to merge. Thanks again.

Also, please consider sharing your tool https://github.com/uswitch/stscreds on the Slack channel to get more feedback or others to test it. That could give it some validation!

**Comment by pingles** [comment]
Yep, that's fine. I'll rework and update here. Good suggestion re:slack, will do that now.

**Comment by jlevy** [comment]
@pingles if you have time to revisit this, appreciate it.

**Comment by pingles** [comment]
I should have time today, thanks for the reminder :-)

**Comment by QuinnyPig** [comment]
@pingles Love to see this get merged this week!

**Comment by FernandoMiguel** [comment]
I've been using aws-vault which uses STS and keychain to keep the keys with lots of success.
Highly recommend it

**Comment by QuinnyPig** [comment]
Going to close this for now until @pingles resurfaces; feel free to reopen.

**Pull request closed without merging**